src/atomicdex/models/qt.global.coins.cfg.proxy.filter.model.cpp

```diff
@@ -0,0 +1,113 @@
+ *                                                                            *
+ * Removal or modification of this copyright notice is prohibited.            *
+ *                                                                            *
+ ******************************************************************************/
+
+#include "atomicdex/models/qt.global.coins.cfg.model.hpp"
+#include "atomicdex/models/qt.global.coins.cfg.proxy.filter.model.hpp"
+#include "atomicdex/utilities/qt.utilities.hpp"
+
+namespace atomic_dex
+{
+    global_coins_cfg_proxy_model::global_coins_cfg_proxy_model(QObject* parent) : QSortFilterProxyModel(parent) {}
+    
+    global_coins_cfg_proxy_model::global_coins_cfg_proxy_model(QObject* parent, CoinType type) :
+        global_coins_cfg_proxy_model(parent)
+    {
+        m_type = type;
+    }
+} // namespace atomic_dex
+
+//! Override
+namespace atomic_dex
+{
+    bool
+    global_coins_cfg_proxy_model::filterAcceptsRow(int source_row, const QModelIndex& source_parent) const
+    {
+        [[maybe_unused]] QModelIndex idx = this->sourceModel()->index(source_row, 0, source_parent);
+        assert(this->sourceModel()->hasIndex(idx.row(), 0));
+
+        if (m_type < CoinType::Disabled)
+        {
+            if (this->sourceModel()->data(idx, global_coins_cfg_model::CoinType) != static_cast<int>(m_type))
+            {
+                return false;
+            }
+            if (this->sourceModel()->data(idx, global_coins_cfg_model::TickerRole).toString() == "All")
+            {
+                return false;
+            }
+        }
+        else if (m_type == CoinType::Disabled)
+        {
+            if (this->sourceModel()->data(idx, global_coins_cfg_model::CurrentlyEnabled).toBool())
+            {
+                return false;
+            }
+        }
+        
+        //! Then use the filter by name
+        return QSortFilterProxyModel::filterAcceptsRow(source_row, source_parent);
+    }
+
+    bool
+    global_coins_cfg_proxy_model::lessThan(const QModelIndex& source_left, const QModelIndex& source_right) const
+    {
+        int      role       = sortRole();
+        QVariant left_data  = sourceModel()->data(source_left, role);
+        QVariant right_data = sourceModel()->data(source_right, role);
+
+        switch (static_cast<global_coins_cfg_model::CoinsRoles>(role))
+        {
+        case global_coins_cfg_model::CoinsRoles::NameRole:
+        {
+            QString left_coin  = left_data.toString();
+            QString right_coin = right_data.toString();
+            if (left_coin == "All")
+            {
+                return true;
+            }
+            return left_coin.toLower() < right_coin.toLower();
+        }
+        default:
+            break;
+        }
+        return false;
+    }
+} // namespace atomic_dex
+
+//! QML API
+namespace atomic_dex
+{
+    void
+    global_coins_cfg_proxy_model::set_all_state(bool checked) noexcept
```

Comment:
`state` is not sufficient here. It should be named `set_all_check_state` at least.